src/components/Experiment/QuickInput/useQuickInputControl.js

```diff
@@ -11,28 +12,51 @@ export default function useQuickInputControl(props) {
   const [selectedInputs, setSelectedInputs] = useState([""]);
   const [selectedTab, setSelectedTab] = useState(0);
 
-  const getTabs = (type = QuickInputType.Image) => {
+  const getTabs = (type = QuickInputType.Image) => {  // TODO: Remove this default
     const sample = {
       id: 'sample-input',
       title: 'Sample inputs',
       component: SampleInputsTab,
       props: {sampleInputs: props.sampleInputs, type: type}
     }
-    const upload = type === QuickInputType.Image ?
-        {id: 'upload-input', title: 'Upload', component: UploadInputsTab} :
-        {id: 'upload-input', title: 'Upload', component: UploadTextInputTab};
-    const input = type === QuickInputType.Image ?
-        {id: 'url-input', title: 'URL', component: URLInputsTab} :
-        {id: 'text-input', title: 'Text', component: TextInputTab};
-
+    const upload = getUploadTabType(type);
+    const input = getInputTabType(type);
     const tabs = [];
 
     if (!props.hideSample) tabs.push(sample);
     if (!props.hideUpload) tabs.push(upload);
-    if (!props.hideUrl) tabs.push(input);
+    if (!props.hideUrl) tabs.push(...input);
 
     return tabs;
   }
+  const getInputTabType = (type) => {
```

Comment:
@ShivanshShalabh This file should have most of the changes that you'll want to reference for your video work.

Reply:
Thanks, Alex, I am trying to integrate the changes into the local repo. I will try to get your latest code and put in my changes to it to avoid any conflict merges.